### PR TITLE
Multi-cluster tests setups

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Create the e2e and integration tests setup for multi cluster suites.

#### Which issue(s) this PR fixes:

Relate to #693

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/kubectl.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+metadata:
+  name: kubectl
+spec:
+  volumes:
+    - name: mk
+      secret:
+        secretName: multikueue
+  containers:
+    - name: kubectl
+      image: bitnami/kubectl:latest
```
Comment (mimowo) [inline hack/kubectl.yaml:12]: I would prefer using the image from the k8s official repo: `registry.k8s.io/kubectl:v1.28.4`. However, maybe we could have some basic ginkgo test, which just lists nodes (or any other resource) on the worker cluster? We need to have the ginkgo tests anyway, so moving the setup to this PR seems to make sense.

File: hack/multikueue-e2e-test.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
+ROOT_DIR=$SOURCE_DIR/..
+export KUSTOMIZE=$ROOT_DIR/bin/kustomize
+export GINKGO=$ROOT_DIR/bin/ginkgo
+export KIND=$ROOT_DIR/bin/kind
+export YQ=$ROOT_DIR/bin/yq
+export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
+export LEADER_KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME}-leader
+export WORKER_KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME}-worker
```
Comment (mimowo) [inline hack/multikueue-e2e-test.sh:29]: I would suggest adding 2 worker clusters in this setup. No reason to go beyond 2 workers, but I think we need 2 to test some important scenarios, for example, 
- to make sure a job does not start at 2 workers at the same time
- the prebuilt workloads are removed from the other workers.

File: test/integration/multikueue/multikueue_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (mimowo) [inline test/integration/multikueue/multikueue_test.go:2]: I feel this is independent of e2e tests, could you send a dedicated PR?

File: test/integration/multikueue/multikueue_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		leaderNs *corev1.Namespace
+		workerNs *corev1.Namespace
+	)
+	ginkgo.BeforeEach(func() {
+		leaderNs = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "multikueue-",
+			},
+		}
+		workerNs = leaderNs.DeepCopy()
```
Comment (mimowo) [inline test/integration/multikueue/multikueue_test.go:40]: I would suggest these tests to also setup the communication between the leader and the worker, creating the appropriate secret, for symmetry with e2e tests. Still, I would suggest this is a dedicated PR.

File: test/integration/multikueue/multikueue_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Multikueue", func() {
+	var (
+		leaderNs *corev1.Namespace
+		workerNs *corev1.Namespace
```
Comment (mimowo) [inline test/integration/multikueue/multikueue_test.go:32]: Similarly as for e2e tests, I would suggest having 2 workers to cover some scenarios, such as removing workloads from the other workers.

Comment (trasc) [reply test/integration/multikueue/multikueue_test.go]: done

File: Makefile [hunk 6]
```diff
@@ -176,6 +178,12 @@ run-test-e2e-%: FORCE
 	@echo Running e2e for k8s ${K8S_VERSION}
 	E2E_KIND_VERSION="kindest/node:v$(K8S_VERSION)" KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) CREATE_KIND_CLUSTER=$(CREATE_KIND_CLUSTER) ARTIFACTS="$(ARTIFACTS)/$@" IMAGE_TAG=$(IMAGE_TAG) ./hack/e2e-test.sh
 
+run-test-multikueue-e2e-%: K8S_VERSION = $(@:run-test-multikueue-e2e-%=%)
+run-test-multikueue-e2e-%: FORCE
+	@echo Running multikueue e2e for k8s ${K8S_VERSION}
+	E2E_KIND_VERSION="kindest/node:v$(K8S_VERSION)" KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) CREATE_KIND_CLUSTER=$(CREATE_KIND_CLUSTER) ARTIFACTS="$(ARTIFACTS)/$@" IMAGE_TAG=$(IMAGE_TAG) ./hack/multikueue-e2e-test.sh
+
+
```
Comment (mimowo) [inline Makefile:186]: nit, remove one empty line

File: hack/mk-leader-cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
```
Comment (mimowo) [inline hack/mk-leader-cluster.yaml:2]: Can we call it manager rather than leader? To align the vocab with KEP, see the naming discussion in the KEP PR: https://github.com/kubernetes-sigs/kueue/pull/1380#discussion_r1409336565.

File: hack/multikueue-e2e-test.sh [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+set -o pipefail
+
+SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
+ROOT_DIR=$SOURCE_DIR/..
+export KUSTOMIZE=$ROOT_DIR/bin/kustomize
+export GINKGO=$ROOT_DIR/bin/ginkgo
+export KIND=$ROOT_DIR/bin/kind
+export YQ=$ROOT_DIR/bin/yq
+export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
+export LEADER_KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME}-leader
```
Comment (mimowo) [inline hack/multikueue-e2e-test.sh:28]: Similar naming comment, I would suggest MANGER, instead of LEADER to align with KEP

File: hack/multikueue-e2e-test.sh [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+				]
+			}
+		], 
+		"restartPolicy":"Never"
+	}
+}'
+kubectl wait pods kubectl-to-1 --for condition=Ready=False --timeout=180s
+kubectl wait pods kubectl-to-2 --for condition=Ready=False --timeout=180s
+
+sleep 10
```
Comment (mimowo) [inline hack/multikueue-e2e-test.sh:232]: Please add a comment why the sleep is needed

File: hack/multikueue-e2e-test.sh [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+					"run", 
+					"testpod",
+					"--image='${E2E_TEST_IMAGE}'"
+				]
+			}
+		], 
+		"restartPolicy":"Never"
+	}
+}'
+kubectl wait pods kubectl-to-1 --for condition=Ready=False --timeout=180s
```
Comment (mimowo) [inline hack/multikueue-e2e-test.sh:229]: Please add a comment why this is needed

File: test/integration/multikueue/multikueue_test.go (hunk 4 above)
Comment (mimowo) [reply test/integration/multikueue/multikueue_test.go]: I proposed to because the suites are independent, so one suite could merge earlier, once ready, but if you prefer to keep them together, I'm ok with that.

File: test/integration/multikueue/multikueue_test.go (hunk 3 above)
Comment (mimowo) [reply test/integration/multikueue/multikueue_test.go:2]: I proposed to because the suites are independent, so one suite could merge earlier, once ready, but if you prefer to keep them together, I'm ok with that.

File: hack/multikueue-e2e-test.sh (hunk 2 above)
Comment (mimowo) [reply hack/multikueue-e2e-test.sh]: I think it is a fair concern, I just tested locally creating 3 clusters and it worked, but I see this might be an issue. If you had to increase the limits to run the tests, I think we could add a comment referring to the potential need to increase the limits. Not clear how often does this happen to me at this moment.

File: Makefile (hunk 6 above)
Comment (trasc) [reply Makefile]: done

File: test/integration/multikueue/multikueue_test.go (hunk 3 above)
Comment (trasc) [reply test/integration/multikueue/multikueue_test.go:2]: it's simpler in one PR

File: hack/kubectl.yaml (hunk 1 above)
Comment (trasc) [reply hack/kubectl.yaml]: ginkgo based tests added

File: hack/multikueue-e2e-test.sh (hunk 2 above)
Comment (trasc) [reply hack/multikueue-e2e-test.sh]: There is a comment in the most likely place to fail (when waiting for kueue to start).

File: hack/mk-leader-cluster.yaml (hunk 7 above)
Comment (trasc) [reply hack/mk-leader-cluster.yaml]: done

File: hack/multikueue-e2e-test.sh (hunk 8 above)
Comment (trasc) [reply hack/multikueue-e2e-test.sh]: done

File: hack/multikueue-e2e-test.sh (hunk 9 above)
Comment (trasc) [reply hack/multikueue-e2e-test.sh]: removed in favor of ginkgo based tests

File: hack/multikueue-e2e-test.sh (hunk 10 above)
Comment (trasc) [reply hack/multikueue-e2e-test.sh]: removed in favor of ginkgo based tests

File: hack/mk-manager-cluster.yaml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  apiServerAddress: "172.17.0.1"
```
Comment (alculquicondor) [inline hack/mk-manager-cluster.yaml:4]: where is this coming from? Why is it the same IP as the worker?

File: hack/mk-manager-cluster.yaml
Comment (alculquicondor) [inline hack/mk-manager-cluster.yaml:1]: maybe let's create a folder "multikueue" and name this `manager-cluster.kind.yaml`

File: test/integration/multikueue/multikueue_test.go [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Multikueue", func() {
+	var (
+		managerNs *corev1.Namespace
+		worker1Ns *corev1.Namespace
+		worker2Ns *corev1.Namespace
```
Comment (alculquicondor) [inline test/integration/multikueue/multikueue_test.go:38]: what are these namespaces for?

File: test/integration/multikueue/multikueue_test.go [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+		ginkgo.DescribeTable("Create workload in cluster", func(c *cluster, ns **corev1.Namespace) {
+			wl := utiltesting.MakeWorkload("wl", (*ns).Name).Obj()
+			gomega.Expect(c.client.Create(c.ctx, wl)).To(gomega.Succeed())
+		},
+			ginkgo.Entry("manager", &managerCluster, &managerNs),
+			ginkgo.Entry("worker1", &worker1Cluster, &worker1Ns),
+			ginkgo.Entry("worker2", &worker2Cluster, &worker2Ns),
+		)
+
+		ginkgo.DescribeTable("Cluster kubeconfig propagation", func(c *cluster, key string, ns **corev1.Namespace) {
```
Comment (alculquicondor) [inline test/integration/multikueue/multikueue_test.go:90]: What is this testing?

File: hack/multikueue-e2e-test.sh
Comment (alculquicondor) [inline hack/multikueue-e2e-test.sh:1]: There is a lot of repeated code. Could you put the functions in a file that can be shared with the single e2e tests?

File: test/mke2e/suite_test.go [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+	managerClusterName := os.Getenv("MANAGER_KIND_CLUSTER_NAME")
+	gomega.Expect(managerClusterName).NotTo(gomega.BeEmpty(), "MANAGER_KIND_CLUSTER_NAME should not be empty")
+
+	worker1ClusterName := os.Getenv("WORKER1_KIND_CLUSTER_NAME")
+	gomega.Expect(worker1ClusterName).NotTo(gomega.BeEmpty(), "WORKER1_KIND_CLUSTER_NAME should not be empty")
+
+	worker2ClusterName := os.Getenv("WORKER2_KIND_CLUSTER_NAME")
+	gomega.Expect(worker1ClusterName).NotTo(gomega.BeEmpty(), "WORKER2_KIND_CLUSTER_NAME should not be empty")
+
+	k8sManagerClient, _ = util.CreateClientUsingCluster("kind-" + managerClusterName)
```
Comment (alculquicondor) [inline test/mke2e/suite_test.go:66]: Don't silence errors (I assume that's what `_` is catching).

File: test/mke2e/e2e_test.go [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+		ginkgo.DescribeTable("Create workload in cluster", func(c *client.Client, ns **corev1.Namespace) {
+			wl := utiltesting.MakeWorkload("wl", (*ns).Name).Obj()
+			gomega.Expect((*c).Create(ctx, wl)).To(gomega.Succeed())
+		},
+			ginkgo.Entry("manager", &k8sManagerClient, &managerNs),
+			ginkgo.Entry("worker1", &k8sWorker1Client, &worker1Ns),
+			ginkgo.Entry("worker2", &k8sWorker2Client, &worker2Ns),
+		)
+
+		ginkgo.DescribeTable("Cluster kubeconfig propagation", func(c *client.Client, key string, ns **corev1.Namespace) {
```
Comment (alculquicondor) [inline test/mke2e/e2e_test.go:70]: Same question, what is this testing?

It looks like it's creating a client based on the secret. But this is not going through the manager cluster in any way. So we are just testing that we can create clients?

File: hack/mk-manager-cluster.yaml
Comment (trasc) [reply hack/mk-manager-cluster.yaml:1]: done

File: hack/multikueue-e2e-test.sh
Comment (trasc) [reply hack/multikueue-e2e-test.sh:1]: done

File: test/mke2e/e2e_test.go (hunk 15 above)
Comment (trasc) [reply test/mke2e/e2e_test.go:70]: I'll remove them in #1313, even without them we'l, need some other place holders.

Comment (alculquicondor) [reply test/mke2e/e2e_test.go:70]: Isn't "Create workload in cluster" enough as a placeholder?

Comment (trasc) [reply test/mke2e/e2e_test.go:70]: the second table has a closer behavior to what we'll need in the future as is also uses the created secret.

File: hack/mk-manager-cluster.yaml (hunk 11 above)
Comment (alculquicondor) [reply hack/mk-manager-cluster.yaml]: If its updated at runtime, maybe we can put `"FILLED_AT_RUNTIME"` instead of a specific IP?

File: test/mke2e/e2e_test.go (hunk 15 above)
Comment (alculquicondor) [reply test/mke2e/e2e_test.go:70]: I still don't see the point, but we can proceed, given that you have a WIP for the next PR.

File: hack/multikueue/worker-cluster.kind.yaml [hunk 16]
```diff
@@ -0,0 +1,4 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  apiServerAddress: "172.17.0.1"
```
Comment (tenzen-y) [inline hack/multikueue/worker-cluster.kind.yaml:4]: Is this IP range used by docker as a default? Or, we use this IP range differentely from the default range?

File: test/integration/multikueue/multikueue_test.go [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+
+		w1Kubeconfig, err := worker1Cluster.kubeConfigBytes()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		w2Kubeconfig, err := worker2Cluster.kubeConfigBytes()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		managerMultikueueSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "multikueue",
```
Comment (tenzen-y) [inline test/integration/multikueue/multikueue_test.go:62]: Is this Secret static name? Or we just have temporary name until the multikueue is implemented?

File: test/integration/multikueue/multikueue_test.go [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+				Namespace: managerNs.Name,
+			},
+			Data: map[string][]byte{
+				"worker1.kubeconfig": w1Kubeconfig,
+				"worker2.kubeconfig": w2Kubeconfig,
```
Comment (tenzen-y) [inline test/integration/multikueue/multikueue_test.go:67]: I'm wondering if we should use the JSON/YAML form for extensibility here.
cc: @alculquicondor

File: test/integration/multikueue/multikueue_test.go [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(managerCluster.ctx, managerCluster.client, managerNs)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(worker1Cluster.ctx, worker1Cluster.client, worker1Ns)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(worker2Cluster.ctx, worker2Cluster.client, worker2Ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.When("Using multiple clusters", func() {
+		ginkgo.DescribeTable("Create workload in cluster", func(c *cluster, ns **corev1.Namespace) {
```
Comment (tenzen-y) [inline test/integration/multikueue/multikueue_test.go:81]:
```golang
		ginkgo.DescribeTable("Create workload in cluster", func(c *cluster, ns **corev1.Namespace) {
```

Why do we need to pass `ns` as typed `**corev1.Namespace`?
IIUC, the following codes wouldn't modify the namespace.

File: test/mke2e/e2e_test.go
Comment (tenzen-y) [inline test/mke2e/e2e_test.go:1]: Let's rename `mke2e` to `multikueuee2e`, the same as other places.
Or, we may be able to re-organize e2e directory the similar to the integration tests like `e2e/multikueue`.

File: test/mke2e/e2e_test.go [hunk 20]
```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
Comment (tenzen-y) [inline test/mke2e/e2e_test.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: test/mke2e/suite_test.go [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+	k8sManagerClient client.Client
+	k8sWorker1Client client.Client
+	k8sWorker2Client client.Client
+	ctx              context.Context
+)
+
+const (
+	Timeout  = time.Minute
+	Interval = time.Millisecond * 250
+)
```
Comment (tenzen-y) [inline test/mke2e/suite_test.go:43]: Can we commonize these variables in 

https://github.com/kubernetes-sigs/kueue/blob/74377c7eb3e37db4703538abbfb9432718be4097/test/util/constants.go
?

File: test/mke2e/e2e_test.go
Comment (trasc) [reply test/mke2e/e2e_test.go:1]: done

File: test/mke2e/e2e_test.go (hunk 20 above)
Comment (trasc) [reply test/mke2e/e2e_test.go]: done

File: test/mke2e/suite_test.go (hunk 21 above)
Comment (trasc) [reply test/mke2e/suite_test.go]: removed

File: test/e2e/multikueue/e2e_test.go [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+var _ = ginkgo.Describe("MultiKueue", func() {
+	var (
+		managerNs *corev1.Namespace
+		worker1Ns *corev1.Namespace
+		worker2Ns *corev1.Namespace
+	)
+	ginkgo.BeforeEach(func() {
+		managerNs = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "multikueue-",
```
Comment (mimowo) [inline test/e2e/multikueue/e2e_test.go:44]: This can result in different namespaces assigned to manager and workers. Can we instead use for workers the same namespace as for manager? So that the integration tests are closer to the real use?

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:44]: This is changed in 1313, the magers namespace is created with `GenarateName` , and the worker namespaces are created with `Name` set to the name got by the manager ns.

Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:44]: I see, the approach with `GenerateName`, then `Name` is what I was thinking about too. 

Since 1313 PR is very large it makes sense to prepare the changes already in this PR. IIUC the purpose of this PR is making the 1313 as small as possible, to make review easier.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:44]: done

File: test/e2e/multikueue/e2e_test.go [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sManagerClient, managerNs)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker1Client, worker1Ns)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker2Client, worker2Ns)).To(gomega.Succeed())
+	})
+	ginkgo.When("Using multiple clusters", func() {
+		ginkgo.DescribeTable("Cluster kubeconfig propagation", func(c *client.Client, key string, ns **corev1.Namespace) {
```
Comment (mimowo) [inline test/e2e/multikueue/e2e_test.go:71]: Can we have `*corev1.Namespace` instead of `**corev1.Namespace`? And analogously in integration tests.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: no, https://github.com/kubernetes-sigs/kueue/pull/1360#discussion_r1423509135

Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:71]: I think it makes it hard to reason about the test. Maybe we could wrap the body of the test in a regular function, and just call twice for both workers?

Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:71]: In particular, I think we we use two `Entry`, then we effectively have two tests, so we run the `BeforeEach` twice. It seems, the intention is to run the `BeforeEach` once, then check kubeconfig propagation for both workers. Unless I'm missing something, I see no point running the `BeforeEach` twice.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: I can't see how is this hard to read.

We will remove these placeholder test when we add some real ones in 1313.

Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:71]: This ** just looks scary, but maybe it's just me :). 

But also that the `BeforeEach` sets up both workers, but in each test we only test one worker, this is counter-intuitive, and wasteful. I would prefer to have a function called twice from within a single `It`, once per worker. This would also eliminate the need for `**` :) 

Anyway, it's not blocking to me, if @alculquicondor and @tenzen-y are happy.

File: hack/multikueue/manager-cluster.kind.yaml
Comment (alculquicondor) [inline hack/multikueue/manager-cluster.kind.yaml:1]: do we even need this file?

File: test/e2e/multikueue/e2e_test.go (hunk 23 above)
Comment (alculquicondor) [reply test/e2e/multikueue/e2e_test.go:71]: Do we even care about different Namespaces for the following PR?

We will probably need the same namespace across all clusters. So I would just remove the namespace altogether here and use `default`.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: > Do we even care about different Namespaces for the following PR?

Yes, I'd prefer to have every test run in it's own set of namespaces.

File: hack/multikueue/manager-cluster.kind.yaml
Comment (trasc) [reply hack/multikueue/manager-cluster.kind.yaml:1]: yes, at least for symmetry when calling `cluster_create`.

File: test/e2e/multikueue/e2e_test.go (hunk 23 above)
Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:71]: I guess a dedicated namespace isn't  required, I'm fine either way. I can see we create generated namespaces for regular e2e tests. I guess one benefit could be to reuse your custom cluster for e2e test so that the tests don't conflict with other stuff.

As for the use of `Entry` I still don't see a need for them, and repeating the setup to test per-worker. I would prefer to have the setup done once and check for both workers. E2e tests are generally costly so it seems to make sense to care about running time, espectially if it does not impact (or even improves) readability.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: This is not the only ginkgo table test in the project.

Comment (mimowo) [reply test/e2e/multikueue/e2e_test.go:71]: Yeah, I'm not saying "never use table tests", it just feels unnecessary in this context, because it results in repeating the e2e setup per-worker while testing a single worker.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: https://github.com/kubernetes-sigs/kueue/pull/1360#discussion_r1422042617

Comment (alculquicondor) [reply test/e2e/multikueue/e2e_test.go:71]: > Yes, I'd prefer to have every test run in it's own set of namespaces.

But this is running in a different namespace for each cluster. I don't see the point of this test. Even if a placeholder, it should be in a form that will be similar to the incoming test.

Comment (trasc) [reply test/e2e/multikueue/e2e_test.go:71]: The namespaces and the ability to use the kubeconfig secret are used in all the new tests that will be added.

Comment (alculquicondor) [reply test/e2e/multikueue/e2e_test.go:71]: Let's proceed, given that the follow up PR is already in the pipeline.

/approve
/lgtm
